src/parse.rs

```diff
@@ -49,6 +49,58 @@ impl Error for ParseError {
     }
 }
 
+impl<'a> MimeIter<'a> {
+    /// A new iterator over mimes or media types
+    pub fn new(s: &'a str) -> Self {
+        Self {
+            pos: 0,
+            source: s,
+        }
+    }
+}
+
+impl<'a> Iterator for MimeIter<'a> {
+    type Item = Result<Mime, &'a str>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let start = self.pos;
+        let len = self.source.bytes().len();
+
+        if start >= len {
+            return None
+        }
+
+        match parse(&self.source[start ..len]) {
+            Ok(value) => {
+                self.pos = len;
+                Some(Ok(value))
+            }
+            Err(ParseError::InvalidToken { pos, .. }) => {
+                if pos == 0 {
+                    self.pos += 1;
+                    return self.next()
+                }
+                let slice = &self.source[start .. start + pos];
+                return match parse(slice) {
```

Comment:
This _kinda_ looks similar to the above, that it will just skip `pos` bytes and try again, right?

Reply:
This tries to parse the slice between the current position and `pos` (not skip, but only deal with). In `application/json, text/html` it would be the position of `,` and the slice here becomes `application/json`